### PR TITLE
Introduce settings page and more options

### PR DESCRIPTION
**Pull request by stklcode**
This PR covers multiple issues and improvements. Feel free to edit or add.

Most significant change is the introduction of a new settings page. The widget backview is reduced to the two widget-related options (about top lists). I added a link to jump directly to the new page to not annoy existing users that are used to the old config panel.

Second I introduced three new checkboxes to control the `skip_tracking` behavior. I added the existing blacklist option to this group alongside with "logged in users", "feed" and "search". This partially covers #103, however the remaining options are not controllable yet. Might be combined with @2ndkauboy's solution to allow total control for pro users.
Especially the user agent filter with the RegEx might be suitable to configure here, but this is more of and advanced feature and not that easy to understand.

With the new page, the role filter (#106) can be integrated smoothly.
PR #72 is somewhat obsoleted or has to be adapted to the new method.

Finally two screenshots of the results:
statify_106_screenshot2
statify_106_screenshot1

### CODE REVIEW

**Review by patrickrobrecht — COMMENTED** [review]
Looks quite good. I've just added a commit fixing the text domain and changing some of the descriptions. I'm wondering whether the settings title can be put into `<label>` for the corresponding `input`s - `add_settings_field` doesn't seem to have an option for that. WP seems to do that on the WP settings pages (except for checkboxes).

Maybe the "skip tracking" options could be displayed as the options on the WP "Discussion" settings page.

**Comment by stklcode** [comment]
> I'm wondering whether the settings title can be put into `<label>` for the corresponding `input`s - `add_settings_field` doesn't seem to have an option for that.

Sure it does: https://developer.wordpress.org/reference/functions/add_settings_field/

> $args
>(array) (Optional) Extra arguments used when outputting the field.
>
>* 'label_for'
> (string) When supplied, the setting title will be wrapped in a <label> element, its for attribute populated with this value.
>* 'class'
> (string) CSS Class to be added to the <tr> element when the field is output.
>
>Default value: array()

Didn't know that until 5 minutes ago, but it obviously works, see latest commit :smiley:

**Review by patrickrobrecht — APPROVED** [review]

File: inc/class-statify-settings.php [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	/**
+	 * Option for tracking via JS.
+	 *
+	 * @return void
+	 */
+	public static function options_snippet() {
+		?>
+		<input type="checkbox" name="statify[snippet]" id="statify_snippet" value="1" <?php checked( Statify::$_options['snippet'], 1 ); ?>
+			title="<?php esc_attr_e( 'Tracking via JavaScript', 'statify' ); ?>">
+		(<?php esc_html_e( 'Default', 'statify' ); ?>: <?php esc_html_e( 'No' ); ?>)
```
Comment (Zodiac1978) [inline inc/class-statify-settings.php]: We should never use core translations. If they change we lose this translation. For consistency and completeness we should always use our own text domain.

Comment (stklcode) [reply inc/class-statify-settings.php]: Okay, let's keep consistent :+1:

**Comment by stklcode** [comment]
The admin menu is yet limited to the `manage_options` privilega, however the links are shown with `edit_dashboard`. Bad UX when clicking a link ends up in a _permission denied_ page.

https://github.com/pluginkollektiv/statify/blob/a2b533009b9884d77051e53aca22d55ee3bebd22/inc/class-statify-settings.php#L289-L295

I separated the 3 possible permissions:
* See dashboard widget - `edit_dashboard` or filter `statify__user_can_see_stats`
* Edit dashboard widget - `edit_dashboard`
* Edit all settings - `manage_options`

**Pull request merged**